src/hivpy/sexual_behaviour.py

```diff
@@ -162,100 +175,110 @@ def _assign_new_sex_group(self, sex, group, rred, size):
+    def init_risk_factors(self, pop: Population):
+        self.init_rred_personal(pop)
+        pop.init_variable(col.RRED_AGE, 1)  # Placeholder to be changed each time step
+        self.update_rred_age(pop)
+        pop.set_present_variable(col.RRED,
+                                 pop.apply_vector_func([col.RRED_PERSONAL, col.RRED_AGE],
+                                                       self.calc_rred_base))
+        self.init_rred_adc(pop)
+        self.init_rred_diagnosis(pop)
         self.init_rred_population()
-        self.init_rred_art_adherence(pop_data)
-        self.init_rred_balance(pop_data)
+        self.init_rred_art_adherence(pop)
+        self.init_rred_balance(pop)
+
+    def calc_rred_base(self, rred_personal, rred_age):
+        return (self.new_partner_factor *
+                rred_personal *
+                rred_age)
 
-    def update_rred(self, population):
-        self.update_rred_adc(population.data)
-        self.update_rred_balance(population.data)
-        self.update_rred_diagnosis(population.data, population.date)
+    def update_rred(self, population: Population):
+        self.update_rred_adc(population)
+        self.update_rred_balance(population)
+        self.update_rred_diagnosis(population)
         self.update_rred_population(population.date)
-        self.update_rred_age(population.data)
-        self.update_rred_long_term_partnered(population.data)
+        self.update_rred_age(population)
+        self.update_rred_long_term_partnered(population)
         if (self.use_rred_art_adherence):
-            self.update_rred_art_adherence(population.data)
-        population.data[col.RRED] = (self.new_partner_factor *
-                                     population.data[col.RRED_AGE] *
-                                     population.data[col.RRED_ADC] *
-                                     population.data[col.RRED_BALANCE] *
-                                     population.data[col.RRED_DIAGNOSIS] *
-                                     population.data[col.RRED_PERSONAL] *
-                                     self.rred_population *
-                                     population.data[col.RRED_LTP] *
-                                     population.data[col.RRED_ART_ADHERENCE])
-
-    def init_rred_art_adherence(self, pop_data):
-        pop_data[col.RRED_ART_ADHERENCE] = 1
-
-    def update_rred_art_adherence(self, pop_data):
-        if ("art_adherence" in pop_data.columns):
-            indices = selector(pop_data, art_adherence=(operator.lt, self.adherence_threshold))
-            pop_data.loc[indices, col.RRED_ART_ADHERENCE] = self.rred_art_adherence
-
-    def update_rred_age(self, pop_data):
-        over_15s = selector(pop_data, age=(operator.ge, 15))
-        age = pop_data.loc[over_15s, col.AGE]
-        sex = pop_data.loc[over_15s, col.SEX]
+            self.update_rred_art_adherence(population)
+
+        def comined_rred(age, adc, balance, diagnosis, personal, ltp, art):
```

Comment:
I take it this should be `combined_rred`?

Reply:
Yes it should!